src/main/scala/viper/gobra/translator/implementations/translator/ExpressionsImpl.scala

```diff
@@ -95,6 +95,8 @@ class ExpressionsImpl extends Expressions {
         case errors => Violation.violation(s"invalid trigger pattern (${errors.head.readableMessage})")
       }
 
+      case in.Conversion(typ, expr) => goE(expr)
```

Comment:
For which types does this work? The typechecker should check that only those supported types are used with conversion expressions.

Reply:
As stated in the meeting, I am not sure that conversions on structs are properly supported.

Reply:
I think that they are supported. According to the Go spec, if two types have the same underlying type (possibly differing only on struct tags, which are not supported in Gobra at the moment) then they can be converted to each other. This is implemented in the convertibility property as follows
https://github.com/viperproject/gobra/blob/c17026c340cc041e7fd71bb50566b2725234952f/src/main/scala/viper/gobra/frontend/info/implementation/property/Convertibility.scala#L19-L22
Both structures are encoded in the same way at the viper level, as tuples, and field accesses are encoded as tuple accesses. Given that two struct types are only equal if they have the same fields, of the same types, and in the same order, field accesses at the viper level are always well typed.

I added a test for struct types, as suggested by you :)